src/activities/extract-meca.ts

```diff
@@ -176,6 +176,14 @@ export const extractMeca = async (version: VersionedReviewedPreprint): Promise<M
     ...supportingFilesUploads,
   ]);
 
+  // Create source.txt in S3 with sourceUuid as its content
+  const s3PathForSourceTxt = constructEPPVersionS3FilePath('source.txt', version);
```

Comment:
Would it not make more sense to put this when we copy the meca (in `copySourcePreprintToEPP()`)?

Reply:
@scottaubrey I have made the refactor that you suggested and agree that this is much cleaner.